СС-18569 Migration guide - Switch from Yarn to NPM v8 (#1113)
spryker/spryker-docs: docs/scos/dev/front-end-development/migration-guide-switch-to-npm.md

---
title: Migration guide - Switch from Yarn to NPM v8
description: Use this guide to migrate project from Yarn to NPM v8.
template: concept-topic-template
---

## Upgrading to version 8.*

> In this article, you will find instructions on how to upgrade NPM to version 8 in your Spryker project.

Reviewer: ```suggestion
This article provides instructions for migrating from YARN to NPM version 8 in your Spryker project.
```

### Overview

> We've been using Yarn v2 since the beginning of Merchant Portal project to install dependencies in a workspace.

Reviewer: ```suggestion
Yarn v2 has been used since the beginning of the Merchant Portal project to install dependencies in a workspace.
```

- `.yarnrc.yml` file
- `yarn.lock` file

### 3) Build the project

1. Run the following commands to apply the docker changes:

```bash
docker/sdk boot deploy.dev.yml
docker/sdk up
```

2. Regenerate `package-lock.json`: 

```bash
rm -rf package-lock.json
docker/sdk cli npm install
```

3. Run the following commands to build the project using NPM v8:

```bash
rm -rf node_modules && docker/sdk cli rm -rf node_modules
docker/sdk boot deploy.dev.yml
docker/sdk up --build --assets --data
```

{% info_block infoBox "Note" %}
The following commands are deprecated and will work correctly only with the previous versions (using Yarn):

```bash
frontend:yves:install-dependencies
frontend:zed:install-dependencies
frontend:mp:install-dependencies
```

To pass an installation of all (yves, zed, mp) frontend dependencies use a single command: 

```bash
frontend:project:install-dependencies
```
{% endinfo_block %}

{% info_block infoBox "Note" %}
If 'node' and 'npm' are uses locally, make sure their versions are correct:

```bash
node -v
npm -v
```

To update them use the following commands:

```bash
sudo npm -g install npm@8
sudo npm cache clean -f
sudo npm install -g n
sudo n 16

> ```

Reviewer: I think we shouldn't provide commands for updating other tools.
Please add links to the appropriate docs.
